scripts/zkai_d128_cryptographic_backend_gate.py

```
@@ -482,6 +486,53 @@ def source_proof_native_contract(source: dict[str, Any], path: pathlib.Path = PR
 
 
+@functools.lru_cache(maxsize=1)
+def _load_checked_snark_receipt_cached(path_text: str) -> dict[str, Any]:
+    path = pathlib.Path(path_text)
+    payload = load_json(path, layer="external_snark_receipt", field="SNARK receipt evidence")
+    expect_equal(payload.get("schema"), EXPECTED_SNARK_RECEIPT_SCHEMA, "SNARK receipt schema", layer="external_snark_receipt")
+    expect_equal(payload.get("issue"), SNARK_RECEIPT_ISSUE, "SNARK receipt issue", layer="external_snark_receipt")
+    expect_equal(payload.get("source_issue"), SOURCE_ISSUE, "SNARK receipt source issue", layer="external_snark_receipt")
+    expect_equal(payload.get("decision"), EXPECTED_SNARK_RECEIPT_DECISION, "SNARK receipt decision", layer="external_snark_receipt")
+    expect_equal(payload.get("result"), EXPECTED_SNARK_RECEIPT_RESULT, "SNARK receipt result", layer="external_snark_receipt")
+    expect_equal(payload.get("all_mutations_rejected"), True, "SNARK receipt mutation result", layer="external_snark_receipt")
```

Ruff (0.15.12)

[warning] Boolean positional value in function call

(FBT003)